### DISCUSSION TITLE
How to build a half-duplex WebRTC server with Pion?

### DISCUSSION THREAD

**Discussion by RussB-tonal** [pion/webrtc#2187] [Q&A] [answered]
I'm trying to use Pion to build a "half-duplex WebRTC server" (ie: video is only sent from a WebRTC client to the server).  I have looked at the "Pion save-to-disk" example for building this.   

Am I correct that if I had multiple WebRTC clients that were sending video to the server, I would need one "save-to-disk" instance for each active user (ie: the  "save-to-disk" example is NOT multi-threaded)?

**Comment by Sean-Der** [comment, marked as answer]
Correct! WebRTC is a one-to-one connection. For each remote peer you need to make a new `webrtc.PeerConnection` in your code.
